test/functional/feature_onlynet.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test -onlynet configuration option.
+"""
+
+from test_framework.basic_server import (
+    BasicServer,
+    tor_control,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+
+class OnlynetTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.tor_control_server = BasicServer(bind=('127.0.0.1', 0),
+                                              response_generator=tor_control)
+        port = self.tor_control_server.listen_port
+        self.extra_args = [
+            ['-onlynet=ipv4', f'-torcontrol=127.0.0.1:{port}', '-listenonion=1'],
+            ['-onlynet=ipv4', f'-torcontrol=127.0.0.1:{port}', '-listenonion=1', '-onion=127.0.0.1:9050'],
+            ['-onlynet=ipv4', f'-torcontrol=127.0.0.1:{port}', '-listenonion=1', '-proxy=127.0.0.1:9050'],
+        ]
+        self.num_nodes = len(self.extra_args)
+
+    def onion_is_reachable(self, node):
+        for net in self.nodes[node].getnetworkinfo()['networks']:
+            if net['name'] == 'onion':
+                return net['reachable']
+        return False
+
+    def run_test(self):
+        # Node 0 would fail the check without c591379eb03381145c85ee5760ce829be337a749.
+        # Nodes 1 and 2 pass with or without that commit, they are here just to ensure
+        # behavior does not change unintentionally.
```

Comment:
maybe use a docstring and describe the commit, feel free to ignore
```diff
-        # Node 0 would fail the check without c591379eb03381145c85ee5760ce829be337a749.
-        # Nodes 1 and 2 pass with or without that commit, they are here just to ensure
-        # behavior does not change unintentionally.
+        """
+        - The test of node 0 would fail without commit c591379eb0,
+          "tor: respect non-onion -onlynet= for outgoing Tor connections."
+
+        - The tests of nodes 1 and 2 pass with or without that commit;
+          they ensure that behavior does not change unintentionally.
+        """
```

Reply:
Leaving it as is.